src/ledger/plan.md

```diff
@@ -0,0 +1,57 @@
+# Ledger Blueprint Planning
+
+The aim is to restructure this documentation with the aim of providing a guide
+to somebody who might wish to implement the ledger for a compatible node.
+
+To that end, we break down the documentation in various sections. Certain
+things make sense to document in certain ways. We wish to avoid any duplication
+of work already done - for example, the formal specs remain probably the best
+way to document the precise implementation of a pure function. However, we can
+provide guidance for people trying to read the specifications.
+
+It is the fundamental nature of documentation to go out of date. As such, we
+also want to avoid referring to details of specific eras etc (which are in any
+case covered in the formal specs) and instead cover the general principles and
+details needed by all potential implementations.
+
+- Concepts
+  - Blocks
+    - The header/body split
+  - Transactions
+  - Eras
+  - The structure of an epoch
+  - Determinism
+- The ledger state transition
+  - How to read the specs
+    - Old-style semi-formal specs
+    - New-style Agda specifications
+  - Validity
+    - Multi-phase validity
+    - Static vs dynamic checks
+- Ledger interfaces
+  - To the consensus layer
+    - Applying a block
+    - Ticking
+      - On an era boundary
+    - Forecasting
+    - Nonces
+    - The stake distribution
+  - To the mempool
+    - Validating a transaction
+    - Revalidating a transaction
+  - To the CLI
+    - Forecasting the leader schedule
+- Understanding parts of the transition
+  - Non-integral math
+  - Transaction fee calculation
+  - Reward calculation
+- Ledger serialisation
```

Comment:
It was discussed in the past that the `cardano-blueprint` could just include the `conway.cddl` et al as generated by the Haskell implementation (the reference implementation). I can imagine a github action which does the synchronization as the least friction and maybe a good first step here!

CDDL was preferred (over Huddle) by most people I talked to as it is more standardized. I'm actively advertising for using `cuddle` (over the `cddl` gem) though.

About the generated ledger CDDL: Some people mentioned that the structure and comments on rules could be improved (it was perceived better when the CDDL was hand-written directly); maybe possible by retaining information available on the DSL?

Reply:
Yes, the point here wouldn't be to include/replace the CDDL, but to discuss other things about ledger serialisation that may not be obvious, such as the lack of canonical formats (and consequent need to retain the bytes), the fact that transactions are serialised differently in the mempool and in blocks, and talk a bit about the ledger state and how it might want to be decomposed.

Reply:
Cuddle should have been updated recently to allow retaining more comments; I don't know if the CDDL has been regenerated recently to take full advantage of this.